src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -182,26 +182,24 @@ export class CalciteRadioButton implements LabelableComponent {
   };
 
   private clickHandler = (): void => {
-    if (this.labelEl) {
-      return;
-    }
-
     this.toggle();
   };
 
   onLabelClick = (event: CustomEvent): void => {
     if (!this.disabled && !this.hidden) {
       this.uncheckOtherRadioButtonsInGroup();
       const label = event.currentTarget as HTMLCalciteLabelElement;
-      const firstButton = this.rootNode.querySelector<HTMLCalciteRadioButtonElement>(
-        label.for
-          ? `calcite-radio-button[id="${label.for}"]`
-          : `calcite-radio-button[name="${this.name}"]`
-      );
-
-      if (firstButton) {
-        firstButton.checked = true;
-        firstButton.focused = true;
+      const radioButton = label.for
+        ? this.rootNode.querySelector<HTMLCalciteRadioButtonElement>(
+            `calcite-radio-button[id="${label.for}"]`
+          )
+        : label.querySelector<HTMLCalciteRadioButtonElement>(
+            `calcite-radio-button[name="${this.name}"]`
+          );
```

Comment:
Even though its probably safe to assume that if a `for` attribute isn't present on the label, that implies that the label contains the radio-button, but in the case that a `for` attribute is supplied to a radio that it is also wrapping (which is probably an anti-pattern I know) it is likely to break in that particular case.  I remember handling that specifically so that it worked regardless if someone did that or not, but its up to you if you think that's a case worth worrying about.